test/unit/hostgroup_extensions_test.rb

```diff
@@ -21,20 +21,22 @@ class HostgroupExtensionsTest < ActiveSupport::TestCase
 
     test 'child host group inherits salt proxy from child parent' do
       parent = FactoryBot.create :hostgroup
-      child_one = FactoryBot.create :hostgroup, :parent => parent
-      child_one.salt_proxy = @proxy
-      child_one.reload
+      child_one = FactoryBot.create :hostgroup, :parent => parent, :salt_proxy => @proxy
       child_two = FactoryBot.create :hostgroup, :parent => child_one
+      assert_not_nil parent
+      assert_not_nil child_one
+      assert_not_nil child_two
```

Comment:
I would also check that neither `child_two.salt_proxy` nor `child_one.salt_proxy` are nil.